### PR TITLE
[Skia] `Frame` improvements

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): related to #8608, related to #4156, related to #12887

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Feature

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/DirectUI/JoltCollections.h.cs [hunk 1]
```diff
@@ -1586,15 +1587,47 @@ private void ClearHandlers()
...
+
+		public virtual void Insert(int index, T item) => _collection.Insert(index, item);
+
+		public virtual void Clear() => _collection.Clear();
+
+		public bool Contains(T item) => _collection.Contains(item);
+
+		public void CopyTo(T[] array, int arrayIndex) => _collection.CopyTo(array, arrayIndex);
+
+		public IEnumerator<T> GetEnumerator() => _collection.GetEnumerator();
```
Comment (Youssef1313) [inline src/Uno.UI/DirectUI/JoltCollections.h.cs:1614]: Maybe better to make the return type `List<T>.Enumerator` instead of `IEnumerator<T>`?

Comment (MartinZikmund) [reply src/Uno.UI/DirectUI/JoltCollections.h.cs:1614]: The type implements `IList` so I think that is not possible (unless I misunderstood)

Comment (Youssef1313) [reply src/Uno.UI/DirectUI/JoltCollections.h.cs:1614]: I think if there is explicit interface implementation, it's valid to change this one

Comment (MartinZikmund) [reply src/Uno.UI/DirectUI/JoltCollections.h.cs:1614]: There is not explicit implementation, if I am not missing something 🤔 . The explicit one is for `IEnumerator`, but this one is still needed for `IEnumerator<T>` to satisfy `IList<T>` I think

File: doc/articles/controls/Frame.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+uid: Uno.Controls.Frame
+---
+
+# Frame
+
+> [!TIP]
+> This article covers Uno-specific information for Frame. For a full description of the feature and instructions on using it, see [Link text](https://learn.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.frame?view=windows-app-sdk-1.5)
```
Comment (Youssef1313) [inline doc/articles/controls/Frame.md:8]:
```suggestion
> This article covers Uno-specific information for the `Frame` class. For a full description of the feature and instructions on using it, see [Frame class](https://learn.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.frame)
```

File: src/Uno.UI.RuntimeTests/Tests/HotReload/Frame/HRApp/Tests/Given_Frame_DataContext.cs [hunk 3]
```diff
@@ -78,6 +78,7 @@ public async Task Check_Can_Change_Page1_With_DataContext()
 	/// Navigate back to Page1
 	/// </summary>
 	[TestMethod]
+	[Ignore("This test is no longer valid with WinUI Frame #17739")]
```
Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/Tests/HotReload/Frame/HRApp/Tests/Given_Frame_DataContext.cs:81]: Could this be causing HR regressions that are visible to the user?

Comment (MartinZikmund) [reply src/Uno.UI.RuntimeTests/Tests/HotReload/Frame/HRApp/Tests/Given_Frame_DataContext.cs:81]: I have created an issue to adjust the tests, but it technically should not, as the test assumes the same instance, but as long as HR updates the type itself, the new instance should have the changes preserved correctly

File: src/Uno.UI/UI/Xaml/Controls/Frame/Frame.Properties.cs [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+	/// Identifies the BackStackDepth dependency property.
+	/// </summary>
+	public static DependencyProperty BackStackDepthProperty { get; } =
+		DependencyProperty.Register(
+			nameof(BackStackDepth),
+			typeof(int),
+			typeof(Frame),
+			new FrameworkPropertyMetadata(0, (s, e) => ((Frame)s)?.OnBackStackDepthChanged(e)));
+
+	protected virtual void OnBackStackDepthChanged(DependencyPropertyChangedEventArgs e)
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/Frame/Frame.Properties.cs:47]: I think it should be `private protected`?

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Controls/Frame/Frame.Properties.cs:47]: I see now it was existing this way already. Let's keep for now :(

File: src/Uno.UI/UI/Xaml/Controls/Frame/Frame.partial.h.mux.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
+﻿using System;
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/Frame/Frame.partial.h.mux.cs:1]: Missing license header?

File: src/Uno.UI/UI/Xaml/Controls/Frame/Frame.partial.mux.cs [hunk 6]
```diff
@@ -0,0 +1,736 @@
+﻿using System;
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/Frame/Frame.partial.mux.cs:1]: Missing license header?

**Review by Youssef1313 — APPROVED** [review]

**Pull request merged**